### ISSUE TITLE
Attention mask question

### ISSUE THREAD

**Issue by gdewael** [bowang-lab/scGPT#131]
Hi,

A question about your masking pre-training procedure:
Given Figure S1 in your paper, it seems to me that the "to-predict" genes can attend on themselves.
Am I then correct in saying that, to predict the gene expression values of a gene, it is conditioned on not only the CLS tokens and other gene expression bins, but also its own gene expression bin?
Another way of phrasing this is: is your pre-training task essentially just to predict the raw expression value given its binned value and some other gene expression bin values?
It this is the case: would it not be more-beneficial to mask out the gene expression bin of the to-predict genes, making it harder for the model, and forcing it only rely on patterns extracted from other genes?

Or am I wrong and is the gene expression bin input for the to-predict genes also replaced with a [MASK] token (as in BERT-style pre-training)?

**Comment by subercui** [comment]
Hi, thank you for the question. I think your description about "to-predict genes can attend on themselves" is correct. I'd like to clarify and highlight that for these to-predict genes, the gene expression is not input to the model. So the task is to predict the gene expression without knowing the raw or binned expression values
